llm/lorax/README.md

```diff
@@ -0,0 +1,102 @@
+# LoRAX: Multi-LoRA inference server that scales to 1000s of fine-tuned LLMs
+
+<p align="center">
+    <img src="https://github.com/predibase/lorax/blob/main/docs/images/lorax_guy.png" alt="LoRAX" style="width:200px;" />
+</p>
+
+[LoRAX](https://github.com/predibase/lorax) (LoRA eXchange) is a framework that allows users to serve thousands of fine-tuned LLMs on a single GPU, dramatically reducing the cost of serving without compromising on throughput or latency. It works by dynamically loading multiple fine-tuned "adapters" (LoRAs, etc.) on top of a single base model at runtime. Concurrent requests for different adapters can be processed together in a single batch, allowing LoRAX to maintain near linear throughput scaling as the number of adapters increases.
+
+## Setup
+
+First install SkyPilot and check that your cloud credentials are properly set:
+
+```shell
+pip install skypilot
+sky check
+```
+
+## Launch a deployment
+
+Create a YAML configuration file called `lorax.yaml`:
+
+```yaml
+resources:
+  cloud: aws
+  accelerators: A10G:1
+  memory: 32+
+  ports: 
+    - 8080
+
+envs:
+  MODEL_ID: mistralai/Mistral-7B-Instruct-v0.1
+
+run: |
+  docker run --gpus all --shm-size 1g -p 8080:80 -v ~/data:/data \
+    ghcr.io/predibase/lorax:latest \
+    --model-id $MODEL_ID
+```
+
+In the above example, we're asking SkyPilot to provision an AWS instance with 1 Nvidia A10G GPU and at least 32GB of RAM. Once the node is provisioned,
+SkyPilot will launch the LoRAX server using our latest pre-built [Docker image](https://github.com/predibase/lorax/pkgs/container/lorax).
+
+Let's launch our LoRAX job:
+
+```shell
+sky launch -c lorax-cluster lorax.yaml
+```
+
+By default, this config will deploy `Mistral-7B-Instruct`, but this can be overridden by running `sky launch` with the argument `--env MODEL_ID=<my_model>`.
+
+**NOTE:** This config will launch the instance on a public IP. It's highly recommended to secure the instance within a private subnet. See the [Advanced Configurations](https://skypilot.readthedocs.io/en/latest/reference/config.html#config-yaml) section of the SkyPilot docs for options to run within VPC and setup private IPs.
+
+## Prompt LoRAX w/ base model
+
+In a separate window, obtain the IP address of the newly created instance:
+
+```shell
+sky status --ip lorax-cluster
+```
+
+Now we can prompt the base model deployment using a simple REST API:
+
+```shell
+IP=$(sky status --ip lorax-cluster)
+
+curl http://$IP:8080/generate \
+    -X POST \
+    -d '{"inputs": "[INST] Natalia sold clips to 48 of her friends in April, and then she sold half as many clips in May. How many clips did Natalia sell altogether in April and May? [/INST]", "parameters": {"max_new_tokens": 64}}' \
+    -H 'Content-Type: application/json'
```

Comment:
nit: we can probably make the post data multiple lines to make it easier to read like the following: https://github.com/skypilot-org/skypilot/blob/c0d23312b0d9808d6b2b849029324e28da35caf4/llm/mixtral/README.md?plain=1#L31-L35